src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -0,0 +1,487 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Resources
+{
+    using System;
+    using System.IO;
+    using System.Globalization;
+    using System.Collections;
+    using System.Text;
+    using System.Threading;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    public sealed class ResourceReader : IDisposable
+    {
+
+        private const int ResSetVersion = 2;
+        private const int ResourceTypeCodeString = 1;
+        private const int ResourceManagerMagicNumber = unchecked((int)0xBEEFCACE);
+        private const int ResourceManagerHeaderVersionNumber = 1;
+
+        private BinaryReader _store;    // backing store we're reading from.
+        private long _nameSectionOffset;  // Offset to name section of file.
+        private long _dataSectionOffset;  // Offset to Data section of file.
+
+        private int[] _namePositions; // relative locations of names
+        private Type[] _typeTable;    // Lazy array of Types for resource values.
+        private int[] _typeNamePositions;  // To delay initialize type table
+        private int _numResources;    // Num of resources files, in case arrays aren't allocated.      
+        private int _version;
+
+
+        public ResourceReader(Stream stream)
+        {
+            if (stream == null)
+                throw new ArgumentNullException("stream");
+            if (!stream.CanRead)
+                throw new ArgumentException(SR.Argument_StreamNotReadable);
+
+            _store = new BinaryReader(stream, Encoding.UTF8);
+
+            ReadResources();
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        private unsafe void Dispose(bool disposing)
+        {
+            if (_store != null)
+            {
+
+                if (disposing)
+                {
+                    // Close the stream in a thread-safe way.  This fix means 
+                    // that we may call Close n times, but that's safe.
+                    BinaryReader copyOfStore = _store;
+                    _store = null;
+                    if (copyOfStore != null)
+                        copyOfStore.Dispose();
+                }
+                _store = null;
+                _namePositions = null;
+
+            }
+        }
+        private void SkipString()
```

Comment:
I can see some other inconsistent formatting where some methods have 2 blank lines and others have 0. The should all have 1. I'm fine with these being cleaned up separate from your split.